### DISCUSSION TITLE
new BlazorExample error

### DISCUSSION THREAD

**Discussion by uwefms** [MarimerLLC/csla#3656] [Questions]
Hi,
 
   when trying to explore the new .net8 BlazorExample I get the following error when trying to add a person:

crit: Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Authorization requires a cascading parameter of type Task<AuthenticationState>. Consider using CascadingAuthenticationState to supply this.
System.InvalidOperationException: Authorization requires a cascading parameter of type Task<AuthenticationState>. Consider using CascadingAuthenticationState to supply this.
   at Microsoft.AspNetCore.Components.Authorization.AuthorizeViewCore.OnParametersSetAsync()
   at Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
   at Microsoft.AspNetCore.Components.ComponentBase.RunInitAndSetParametersAsync()

  regards
   
     Uwe

**Comment by rockfordlhotka** [comment]
Yes, I'm still trying to resolve that issue - learning more details about authentication/authorization in Blazor 8.

**Reply by uwefms** [reply]
Hi Rocky,

  did some more research with the R24012202 version.

There was a discussion on this topic in

https://github.com/dotnet/aspnetcore/issues/50951

Steve suggested to add 

builder.Services.AddCascadingAuthenticationState();

in program.cs. I did this and also added

	[CascadingParameter]
	private Task<AuthenticationState>? authenticationState { get; set; }

to Counter.cs and EditPerson.cs. Calling EditPerson for the first time works without error but changing to Counter produces the following timeout error when calling await StateManager.InitialiteAsync()


TimeoutException: The operation has timed out.
Csla.Blazor.State.StateManager.WaitForState(TimeSpan timeout)
Csla.Blazor.State.StateManager.GetState(TimeSpan timeout)
Csla.Blazor.State.StateManager.InitializeAsync(TimeSpan timeout)
Csla.Blazor.State.StateManager.InitializeAsync()
BlazorExample.Client.Components.Pages.Counter.OnInitializedAsync() in Counter.razor
+
	protected override async Task OnInitializedAsync()
    {
		if (authenticationState is not null)
		{
			// Every page _must_ initialize the state manager
			await StateManager.InitializeAsync();
		}
    }

  HTH    
    regards
      Uwe

**Reply by rockfordlhotka** [reply]
That is fantastic @uwefms - thank you for that!

The timeout is due to an oversight on my part with navigating from one wasm page to another. I'll (hopefully) figure that out today and have a solution.

**Update** maybe less of an oversight on my part, and more of an unexpected behavior. It appears that a new DI scope is being created as we navigate from one wasm page to another. I surely didn't expect that!

**Reply by rockfordlhotka** [reply]
The solution (at least for now) appears to be that wasm pages need to have their render mode set to prevent server prerendering.

```c#
// counter page
@rendermode @(new InteractiveAutoRenderMode(prerender: false))

// personedit page
@rendermode @(new InteractiveWebAssemblyRenderMode(prerender: false))
```

**Reply by rockfordlhotka** [reply]
https://github.com/MarimerLLC/csla/pull/3662

**Comment by ctwoodwa** [comment]
I finally have some time to get back to learning CSLA.  When I open the latest Blazor sample project the version is 8.0.0-R24010204, but this version is not listed on Nuget.  the latest version is 8.0.0-R24010305, but if I use this version there are a couple of compile issues.  Any recommendations on how to proceed?

**Comment by rockfordlhotka** [comment]
@ctwoodwa I was able to update to the R24010305 release and have the app work (with the exception of the wasm page as noted above). There were some compile warnings, but they went away on a subsequent rebuild, so I think it was due to old stuff in the obj folders.

**Reply by ctwoodwa** [reply]
I can confirm it works for me too.

**Comment by rockfordlhotka** [comment]
I just merged a PR that updates the BlazorExample, so I think it is working now in `main`.
